### PR TITLE
update the dependencies to latest version

### PR DESCRIPTION
**Pull request by stevefan1999-personal**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,17 +11,18 @@ categories = ["embedded", "no-std", "network-programming"]
 readme = "README.md"
 
 [dependencies]
-sha1 = "0.6"
-heapless = "0.5"
-byteorder = { version = "1.4", default-features = false }
-httparse = { version = "1.4", default-features = false }
-rand_core = "0.6"
+sha1 = "0.10.1"
```
Comment (ninjasource) [inline Cargo.toml:14]: Needs to be compatible with no_std as follows:
```toml
sha1 = { version = "0.10.1", default-features = false }
```

File: Cargo.toml [hunk 2]
```diff
@@ -11,17 +11,18 @@ categories = ["embedded", "no-std", "network-programming"]
...
+heapless = "0.7.14"
+byteorder = { version = "1.4.3", default-features = false }
+httparse = { version = "1.7.1", default-features = false }
+rand_core = "0.6.3"
```
Comment (ninjasource) [inline Cargo.toml:18]: Same with sha1 above

**Review by ninjasource — CHANGES_REQUESTED** [review]

**Comment by ninjasource** [comment]
Thanks for your PR!

File: Cargo.toml (hunk 2 above)
Comment (stevefan1999-personal) [reply Cargo.toml:18]: I check it with `cargo feature`. All of them should be no_std by default.

**Review by ninjasource — APPROVED** [review]
All good, thanks for your changes!

**Pull request merged**